src/block/stdio_executor.rs

```diff
@@ -233,16 +241,16 @@ impl<B: Backend> StdIoBackend<B> {
                 for (data_addr, data_len) in request.data() {
                     mem.read_exact_from(*data_addr, &mut self.inner, *data_len as usize)
                         .map_err(Error::Read)?;
-                    bytes_from_dev += data_len;
+                    bytes_from_dev = bytes_from_dev
+                        .checked_add(*data_len)
+                        .ok_or_else(|| Error::Overflow)?;
```

Comment:
I think we can use `.ok_or(Error::Overflow)` directly here instead of `ok_or_else`.